compiler/rustc_typeck/src/check/callee.rs

```diff
@@ -241,7 +241,9 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                         if let ty::Ref(r, _, mutbl) = method.sig.inputs()[0].kind() {
                             (r, mutbl)
                         } else {
-                            span_bug!(call_expr.span, "input to call/call_mut is not a ref?");
+                            // The `fn`/`fn_mut` lang item is ill-formed, which should have
+                            // caused an error elsewhere.
+                            return None;
```

Comment:
Maybe make this a `delay_span_bug!`? That will cause a `span_bug!` only when no other errors are reported.

Reply:
Sounds good; fixed.